iceoryx_hoofs/source/posix_wrapper/named_pipe.cpp

```diff
@@ -0,0 +1,235 @@
+
+    if (sharedMemory.has_error())
+    {
+        std::cerr << "Unable to open shared memory: \"" << convertName(name) << "\" for named pipe \"" << name << "\""
+                  << std::endl;
+        m_isInitialized = false;
+        m_errorValue = (channelSide == IpcChannelSide::CLIENT) ? IpcChannelError::NO_SUCH_CHANNEL
+                                                               : IpcChannelError::INTERNAL_LOGIC_ERROR;
+        return;
+    }
+
+    m_sharedMemory.emplace(std::move(sharedMemory.value()));
+    m_messages = static_cast<MessageQueue_t*>(m_sharedMemory->getBaseAddress());
+
+    if (channelSide == IpcChannelSide::SERVER)
+    {
+        new (m_messages) MessageQueue_t();
+    }
+    m_isInitialized = true;
+}
+
+NamedPipe::NamedPipe(NamedPipe&& rhs) noexcept
+{
+    *this = std::move(rhs);
+}
+
+NamedPipe& NamedPipe::operator=(NamedPipe&& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        IOX_DISCARD_RESULT(destroy());
+    }
+    CreationPattern_t::operator=(std::move(rhs));
+
+    m_sharedMemory = std::move(rhs.m_sharedMemory);
+    m_messages = std::move(rhs.m_messages);
+    rhs.m_messages = nullptr;
+
+    return *this;
+}
+
+NamedPipe::~NamedPipe() noexcept
+{
+    IOX_DISCARD_RESULT(destroy());
+}
+
+IpcChannelName_t NamedPipe::convertName(const IpcChannelName_t& name) noexcept
+{
+    return IpcChannelName_t(
+        cxx::TruncateToCapacity,
+        cxx::concatenate(NAMED_PIPE_PREFIX, (name.c_str()[0] == '/') ? *name.substr(1) : name).c_str());
+}
+
+cxx::expected<IpcChannelError> NamedPipe::destroy() noexcept
+{
+    if (m_isInitialized)
+    {
+        m_isInitialized = false;
+        m_errorValue = IpcChannelError::NOT_INITIALIZED;
+
+        m_sharedMemory.reset();
+        m_messages = nullptr;
+    }
+    return cxx::success<>();
+}
+
+cxx::expected<bool, IpcChannelError> NamedPipe::isOutdated() noexcept
+{
+    return cxx::success<bool>(false);
+}
+
+cxx::expected<bool, IpcChannelError> NamedPipe::unlinkIfExists(const IpcChannelName_t& name) noexcept
+{
+    constexpr int ERROR_CODE = -1;
+    auto unlinkCall =
+        posixCall(shm_unlink)(convertName(name).c_str()).failureReturnValue(ERROR_CODE).ignoreErrnos(ENOENT).evaluate();
+    if (!unlinkCall.has_error())
+    {
+        return cxx::success<bool>(unlinkCall->errnum != ENOENT);
+    }
+    else
+    {
+        return cxx::error<IpcChannelError>(IpcChannelError::INTERNAL_LOGIC_ERROR);
+    }
+}
+
+cxx::expected<IpcChannelError> NamedPipe::send(const std::string& message) const noexcept
+{
+    return timedSend(message, BLOCKING_TIMEOUT);
+}
+
+cxx::expected<IpcChannelError> NamedPipe::timedSend(const std::string& message,
+                                                    const units::Duration& timeout) const noexcept
+{
+    if (!m_isInitialized)
+    {
+        return cxx::error<IpcChannelError>(IpcChannelError::NOT_INITIALIZED);
+    }
+
+    if (message.size() > MAX_MESSAGE_SIZE)
```

Comment:
My expectation was that we compare with `maxMsgSize` here which is a c'tor argument but was never stored. Can you please leave a comment in the c'tor why `maxMsgSize` is passed but not used? Same for `maxMsgNumber`.